fork-management-for-infra-providers/README.md

```diff
@@ -0,0 +1,148 @@
+  - Generate base manifests by running scripts of the core repo
+  - Customize base manifests with kustomization. [Example](https://github.com/giantswarm/cluster-api-provider-cloud-director-app/blob/adfc74115d9b7aeda60d885500122be3fb434d2e/Makefile.custom.mk#L19)
+  - Update manifests under `helm` folder for the app. [Example](https://github.com/giantswarm/cluster-api-provider-cloud-director-app/blob/adfc74115d9b7aeda60d885500122be3fb434d2e/Makefile.custom.mk#L28)
+
+When it is necessary to update the app content, developers are supposed to
+- run make targets manually to update manifests and values in the app repo
+- open a PR in the app repo
+
+### Image Management
+
+For some providers, we retag upstream images with retagger.
+For some providers like CAPVCD, we build/push images manually.
+
+## Design proposals
+
+### 1. Repo management
+
+#### 1.a Separate repositories for fork and app
+
+We can have two repositories per operator. One is only fork of upstream and the other one is for only bundling it as an app.
+
+#### 1.b One repository for fork and app
+
+We can maintain bundling and customization logic in the same repo.
+
+#### Decision
+
+- `1.a` is the current situation.
+- `1.b` will lead to many conflicts with the upstream repository and it will be hard to automate conflict resolution.
+
+`1.a` is decided.
+
+---
+
+### 2. Branching strategy in fork repo
+
+#### 2.a main vs giantswarm-main
+
+We can maintain two branches in the fork repo. 
+- `main` will be just the same as `main` of upstream repository. This branch can be synced automatically.
+- `giantswarm-main` will contain our changes if necessary. 
+
+The update flow will be
+- First update `main`
+- Then rebase `giantswarm-main`
+
+#### 2.b only main
+
+We can maintain only one `main` branch including our changes and use `git merge` to apply upstream changes.
+
+#### Decision
+
+`2.a` is decided.
+
+---
+
+### 3. Image building
+
+#### 3.a Not building images 
+
+It is not possible to rely on only retagged versions of containers provided by the upstream community since we want to be able to apply our changes to code base.
+
+#### 3.b Building/Pushing images in the fork repo
+
+We can add an image building/pushing mechanism to fork repository.
+
+- This will increase the diff between fork and upstram repo but it will be only some manifests under `.github` or `.circleci` folders.
+- This will allow developers to verify container building while developing features.
+
+#### 3.c Building/Pushing images in the app repo
+
+We can build/push images while syncing the app repo. That works pretty well while consuming upstream repo but it will make less sense when having our own fork. 
+
+#### Decision
+
+`3.b` is decided.
+
+---
+
+### 4. Where to build the images
+
+#### 4.a Github Actions
+
+#### 4.b CircleCI
+
+#### Decision
+
+Since we build/push images via CircleCI for our other repositories, `4.b` is decided.
+
+---
+
+### 5. Sync logic in the app repo
```

Comment:
Syncing upstream changes should be fully automated and require no human intervention, unless we have our changes and there are conflicts.